pkg/test/framework/components/stackdriver/google.go

```diff
@@ -0,0 +1,217 @@
+			IntervalStartTime(startTime.Format(time.RFC3339)).
+			IntervalEndTime(endTime.Format(time.RFC3339)).
+			AggregationCrossSeriesReducer("REDUCE_NONE").
+			AggregationAlignmentPeriod("60s").
+			AggregationPerSeriesAligner("ALIGN_RATE").
+			Filter(fmt.Sprintf("metric.type = %q AND resource.type = %q AND resource.labels.namespace_name = %q", q.metricName, q.resourceType, namespace)).
+			Context(context.Background())
+		resp, err := lr.Do()
+		if err != nil {
+			return nil, err
+		}
+		if resp.HTTPStatusCode != http.StatusOK {
+			return nil, fmt.Errorf("failed to get expected status code from monitoring service, got: %d", resp.HTTPStatusCode)
+		}
+		for _, ts := range resp.TimeSeries {
+			newTS := &monitoringpb.TimeSeries{}
+			if ts.Metric == nil {
+				continue
+			}
+			newTS.Metric = &metric.Metric{}
+			newTS.Metric.Labels = ts.Metric.Labels
+			newTS.Metric.Type = ts.Metric.Type
+			if ts.Resource == nil {
+				continue
+			}
+			newTS.Resource = &monitoredres.MonitoredResource{}
+			newTS.Resource.Type = ts.Resource.Type
+			newTS.Resource.Labels = ts.Resource.Labels
+			ret.TimeSeries = append(ret.TimeSeries, newTS)
+		}
+	}
+
+	return trimMetricLabels(ret), nil
+}
+
+func (s *realStackdriver) ListLogEntries(filter LogType, namespace string) ([]*loggingpb.LogEntry, error) {
+	logName := logNameSuffix(filter)
+	resp, err := s.loggingService.Entries.List(&logging.ListLogEntriesRequest{
+		ResourceNames: []string{fmt.Sprintf("projects/%v", s.projectID)},
+		PageSize:      200,
+		Filter: fmt.Sprintf("timestamp > %q AND logName:%q AND resource.labels.namespace_name=%q",
+			time.Now().Add(-5*time.Minute).Format(time.RFC3339), logName, namespace),
+	}).Context(context.Background()).Do()
+	if err != nil {
+		return nil, fmt.Errorf("unexpected error from the logging backend: %v", err)
+	}
+	if resp.HTTPStatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code from logging service, got: %d", resp.HTTPStatusCode)
+	}
+
+	resppb := loggingpb.ListLogEntriesResponse{
+		Entries: make([]*loggingpb.LogEntry, len(resp.Entries)),
+	}
+	for i, le := range resp.Entries {
+		resppb.Entries[i] = &loggingpb.LogEntry{}
+		resppb.Entries[i].LogName = le.LogName
+		if le.TextPayload != "" {
+			resppb.Entries[i].Payload = &loggingpb.LogEntry_TextPayload{
+				TextPayload: le.TextPayload,
+			}
+		}
+		if le.HttpRequest != nil {
+			resppb.Entries[i].HttpRequest = &ltype.HttpRequest{}
+			resppb.Entries[i].HttpRequest.RequestMethod = le.HttpRequest.RequestMethod
+			resppb.Entries[i].HttpRequest.RequestUrl = le.HttpRequest.RequestUrl
+			resppb.Entries[i].HttpRequest.Status = int32(le.HttpRequest.Status)
+			resppb.Entries[i].HttpRequest.Protocol = le.HttpRequest.Protocol
+		}
+		resppb.Entries[i].Labels = le.Labels
+		resppb.Entries[i].TraceSampled = le.TraceSampled
+	}
+	return trimLogLabels(&resppb, filter), nil
+}
+
+func (s *realStackdriver) ListTrafficAssertions() ([]*edgespb.TrafficAssertion, error) {
```

Comment:
we should file a follow-on issue to remove this from all tests, etc.

Reply:
Cleaned up at https://github.com/istio/istio/pull/31796